src/darwin/Framework/CHIP/MTRAsyncWorkQueue.mm

```diff
@@ -222,18 +292,20 @@ - (void)_postProcessWorkItem:(MTRAsyncWorkItem *)workItem retry:(BOOL)retry
         return;
     }
 
-    // if work item is done (no need to retry), remove from queue and call ready on the next item
-    if (!retry) {
+    if (retry) {
+        MTR_LOG_DEFAULT("MTRAsyncWorkQueue<%@> retry needed for work item %tu", context.description, workItem.uniqueID);
```

Comment:
Minor question: why are you passing `context.description` rather than just `context`? Doesn't "%@" lead to calling `-description` anyway?